test/python/opflow/test_aer_pauli_expectation.py

```diff
@@ -120,6 +121,29 @@ def test_pauli_expect_state_vector(self):
 
         np.testing.assert_array_almost_equal(sampled.eval(), [0, 0, 1, -1], decimal=1)
 
+    def test_pauli_expect_non_hermitian_matrixop(self):
+        """pauli expect state vector with non hermitian operator test"""
+        states_op = ListOp([One, Zero, Plus, Minus])
+
+        op_mat = np.array([[0, 1], [2, 3]])
+        op = MatrixOp(op_mat)
+
+        converted_meas = self.expect.convert(~StateFn(op) @ states_op)
+        sampled = self.sampler.convert(converted_meas)
+
+        np.testing.assert_array_almost_equal(sampled.eval(), [3, 0, 3, 0], decimal=1)
+
+    def test_pauli_expect_non_hermitian_pauliop(self):
+        """pauli expect state vector with non hermitian operator test"""
+        states_op = ListOp([One, Zero, Plus, Minus])
+
+        op = 1j * X
+
+        converted_meas = self.expect.convert(~StateFn(op) @ states_op)
```

Comment:
Using `~` will complex conjugate the observable, which we don't really want
```suggestion
        converted_meas = self.expect.convert(StateFn(op, is_measurement=True) @ states_op)
```

Reply:
I agree with the commit and I changed the expected result from [0, 0, -1j, 1j] to [0, 0, 1j, -1j] to match this modification.